classes/Tools.php

```diff
@@ -202,10 +202,7 @@ public static function redirect($url, $base_uri = __PS_BASE_URI__, Link $link =
             }
 
             $explode = explode('?', $url);
-            // don't use ssl if url is home page
-            // used when logout for example
-            $use_ssl = !empty($url);
```

Comment:
You should remove this line.

Reply:
this line ? : `$explode = explode('?', $url);`